src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,59 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
-    public void getStatistic(String fromFileName, String toFileName) {
+    private static final String COMA_SEPARATOR = ",";
+    private static final int VALUE_INDEX = 1;
+    private static final int OPERATION_TYPE_INDEX = 0;
+
+    protected void getStatistic(String fromFileName, String toFileName) {
+        String[] fileInfo = this.readFromFile(fromFileName);
+        int supplySum = 0;
+        int buySum = 0;
+        for (String oneLine : fileInfo) {
+            String[] stringInfo = oneLine.split(COMA_SEPARATOR);
+            if (stringInfo[OPERATION_TYPE_INDEX].equals("supply")) {
+                supplySum += Integer.parseInt(stringInfo[VALUE_INDEX]);
+            } else {
+                buySum += Integer.parseInt(stringInfo[VALUE_INDEX]);
+            }
+        }
+        String toFileInfo = createReport(supplySum, buySum);
+        this.writeToFile(toFileName, toFileInfo);
+    }
+
+    private String[] readFromFile(String fromFileName) {
+        StringBuilder stringBuilder = new StringBuilder();
+        try (BufferedReader bufferedReader = new BufferedReader(
+                new java.io.FileReader(fromFileName))) {
```

Comment:
It is bad practice to write class package while creating object it will be better to avoid to and write just "new FileReader(fromFileName)"